packages/trie/src/node/extensionOrLeafNodeBase.ts

```diff
@@ -3,23 +3,27 @@ import { RLP } from '@ethereumjs/rlp'
 import { addHexPrefix, removeHexPrefix } from '../util/hex.js'
 import { nibblesTypeToPackedBytes } from '../util/nibbles.js'
 
-import type { Nibbles } from '../types.js'
+import type { Nibbles, RawExtensionNode, RawLeafNode } from '../types.js'
 
-export class Node {
+export abstract class ExtensionOrLeafNodeBase {
   _nibbles: Nibbles
   _value: Uint8Array
-  _terminator: boolean
+  _isLeaf: boolean
```

Comment:
With these changes I suggest we remove the `ExtensionNode` and `LeafNode` completely. We could introduce static constructors here which return the right node type. (`_isLeaf` thus needs to be public in order to determine if it is a leaf/extension node)

Reply:
I went back and forth on this earlier today, but figured it was better to have these classes separate just for naming and clarity purposes, just so that we can at least do `new LeafNode` and `new ExtensionNode` instead of something like `new ExtensionOrLeafNode(isLeaf: true)`. What do feel is the value in having them both be the same class?

Reply:
You are right, the classes are much more clear for readability! Let's not change it.